### PR TITLE
add customClass props to Skeletons

### PR DESCRIPTION
**Pull request by GMartigny**
## Proposed Changes

- Add a `customClass` prop to Skeletons
- Use `<template>` rendering instead of functionnal

### CODE REVIEW

**Comment by jtommy** [comment]
@GMartigny why did you move to template rendering?

**Comment by GMartigny** [comment]
Because functional didn't serve any purpose here and all the rest of the lib is template. But of course, it can be reverted if you think it's out of scope for this PR.

**Comment by jtommy** [comment]
And `custom-class`? You can use class and it will be applied to root element.
Probably with functional is not possible, but you can add reading it from `vnode` (second parameter of render function), for example: `vnode.data.staticClass`

Functional are great in Vue 2 for a component without state and there are benchmarks in the web that show rendering improvements.

**Comment by GMartigny** [comment]
> Probably with functional is not possible, but you can add reading it from vnode (second parameter of render function), for example: vnode.data.staticClass

This is indeed need to [be explicit with functional rendering](https://vuejs.org/v2/guide/render-function.html#Passing-Attributes-and-Events-to-Child-Elements-Components).

> Functional are great in Vue 2 for a component without state and there are benchmarks in the web that show rendering improvements.

Didn't know, thanks for the infos. However, I still find it quite hard to parse for a human.

I took the liberty to spread the whole data coming from the context (`...context.data`) into the component. This allow users to bind `class`, `id` and event listeners. What do you think ?

**Comment by jtommy** [comment]
Ok!

File: src/components/skeleton/Skeleton.vue [hunk 1]
```diff
@@ -55,8 +55,9 @@ export default {
             }))
         }
         return createElement('div', {
+            ...context.data,
             staticClass: 'b-skeleton',
-            class: [ context.props.size, context.props.position, { 'is-animated': context.props.animated } ]
+            class: [ context.data.attrs.class, context.props.size, context.props.position, { 'is-animated': context.props.animated } ]
```
Comment (jtommy) [inline src/components/skeleton/Skeleton.vue:60]: Are you sur is it ok? I usually use .staticClass and not .class

Comment (jtommy) [inline src/components/skeleton/Skeleton.vue:60]: Are you sur is it ok? I usually use .`staticClass` and not .class

**Review by jtommy — CHANGES_REQUESTED** [review]

File: src/components/skeleton/Skeleton.vue (hunk 1 above)
Comment (GMartigny) [reply src/components/skeleton/Skeleton.vue:60]: Do you know the proper way to test this ? 

If I [do ](https://github.com/buefy/buefy/pull/3642/files#diff-6c1222579876270e0ad765a3dc4efc91824ab4e1395b805b74d352d15c2b843bR8-R12):
```js
wrapper = mount(BSkeleton, {
  attrs: {
    class: 'custom-class'
  }
})
```
I don't have `staticClass` set.

Comment (jtommy) [reply src/components/skeleton/Skeleton.vue:60]: I mean in the core code, you should use `context.data.attrs.class` instead of `context.data.attrs.class`

**Comment by QuentinGab** [comment]
Hi !

Is there any update on this @GMartigny  ? It would be cool to be able to pass classes and styles to the component

**Comment by GMartigny** [comment]
Honestly, I don't know how to do it properly in the functional way (especially with testing). Maybe @jtommy or someone else could finish this up.

**Comment by jtommy** [comment]
As i told you https://github.com/buefy/buefy/pull/3642/files#r794054499, did you try it?

**Comment by jtommy** [comment]
@GMartigny any news?

**Comment by GMartigny** [comment]
As I said in my previous comment, I'm not familiar with Vue functionnal rendering. I tried your suggestion, but I'm not able to properly test it. If i use `context.data.attrs.staticClass` I don't have the class correctly set in tests.
